core/src/main/java/org/keycloak/sdjwt/ClaimVerifier.java

```diff
@@ -495,23 +489,17 @@ public Builder withExpCheck() {
         }
 
         public Builder withExpCheck(boolean isCheckOptional) {
-            contentVerifiers.removeIf(verifier -> {
-                return verifier instanceof ClaimVerifier.ExpCheck ||
-                    (verifier instanceof ClaimVerifier.ClaimCheck
-                        && ((ClaimCheck) verifier).getClaimName().equalsIgnoreCase(OID4VCConstants.CLAIM_NAME_EXP));
-            });
+            contentVerifiers.removeIf(verifier ->
+                    isTypeOrNamedCheck(verifier, ClaimVerifier.ExpCheck.class, OID4VCConstants.CLAIM_NAME_EXP));
```

Comment:
you have remove the fallback checks in case that custom-verifiers were added. Please readd these checks for the methods